cli/cmd/check.go

```diff
@@ -150,30 +135,49 @@ func (o *checkOptions) validate() error {
 }
 
 func runChecks(w io.Writer, hc *healthcheck.HealthChecker) bool {
+	var lastCategory healthcheck.CategoryID
+	spin := spinner.New(spinner.CharSets[0], 100*time.Millisecond)
+	spin.Writer = w
+
 	prettyPrintResults := func(result *healthcheck.CheckResult) {
-		checkLabel := fmt.Sprintf("%s: %s", result.Category, result.Description)
+		if lastCategory != result.Category {
+			if lastCategory != "" {
+				fmt.Fprintln(w, "")
+			}
+
+			underline := ""
+			for i := 0; i < len(result.Category); i++ {
+				underline = underline + "-"
+			}
 
-		filler := ""
-		lineBreak := "\n"
-		for i := 0; i < lineWidth-len(checkLabel)-len(okStatus)-len(lineBreak); i++ {
-			filler = filler + "."
+			fmt.Fprintln(w, result.Category)
+			fmt.Fprintln(w, underline)
+
+			lastCategory = result.Category
 		}
 
+		spin.Stop()
 		if result.Retry {
-			fmt.Fprintf(w, "%s%s%s -- %s%s", checkLabel, filler, retryStatus, result.Err, lineBreak)
+			spin.Suffix = fmt.Sprintf("  %s -- %s", result.Description, result.Err)
+			spin.Color("bgBlack", "bold", "fgRed")
```

Comment:
I have a light theme in my console (I know, weird, but I have my reasons :) ) so the spinner doesn't look right. Although I looked a bit into the spinner lib and couldn't find a way to cater for both dark and light themes at the same time :-(

Reply:
@alpeb can you try replacing that `spin.Color(...` line with `spin.Start(...`? maybe the default colors will play better in light and dark themes?